Update how to on pulsar phase (#5315)
gammapy/gammapy: docs/user-guide/howto.rst

    :id: HowToAddPhase
    :title: Add PHASE information to your data

> To do a pulsar analysis, one must compute the pulsar phase of

Reviewer: ```suggestion
To perform a pulsar analysis, one must compute the pulsar phase of
```

each event and put this new information in a new `~gammapy.data.Observation`.
Computing pulsar phases can be done using an external library such as

> [PINT](https://nanograv-pint.readthedocs.io/en/latest/) or
> [Tempo2](https://www.pulsarastronomy.net/pulsar/software/tempo2). A
> [Gammapy Recipe](https://gammapy.github.io/gammapy-recipes/_build/html/index.html)
> showing how to use PINT within the Gammapy framework is available
> [here](https://gammapy.github.io/gammapy-recipes/_build/html/notebooks/pulsar_phase/pulsar_phase_computation.html).

Reviewer: ```suggestion
`PINT <ttps://nanograv-pint.readthedocs.io/en/latest/>`__ or
`Tempo2 <https://www.pulsarastronomy.net/pulsar/software/tempo2>`__. A
`Gammapy Recipe <https://gammapy.github.io/gammapy-recipes/_build/html/index.html>`__
showing how to use PINT within the Gammapy framework is available
`here <https://gammapy.github.io/gammapy-recipes/_build/html/notebooks/pulsar_phase/pulsar_phase_computation.html>`__.
```

Author: Thanks @Astro-Kirsty !

To perform a pulsar analysis, one must compute the pulsar phase of
each event and put this new information in a new `~gammapy.data.Observation`.
Computing pulsar phases can be done using an external library such as

> `PINT <ttps://nanograv-pint.readthedocs.io/en/latest/>`__ or

Reviewer: ```suggestion
`PINT <https://nanograv-pint.readthedocs.io/en/latest/>`__ or
```

    # create a new `EventList`
    table = obs.events.table
    table["PHASE"] = phase

>     new_events = EventList(table)

Reviewer 2: ```suggestion
    events_with_phase = EventList(table)
```
I would suggest to be as explicit as possible.

    # copy the observation in memory, changing the events

>     obs2 = obs.copy(events=new_events, in_memory=True)

Reviewer 2: ```suggestion
    obs_with_phase = obs.copy(events=events_with_phase, in_memory=True)
```

    # The new observation and the new events table can be serialised independently

>     obs2.write("new_obs.fits.gz")
>     obs2.write("events.fits.gz", include_irfs=False)

Reviewer 2: ```suggestion
    obs_with_phase.write("new_obs.fits.gz")
    obs_with_phase.write("events.fits.gz", include_irfs=False)
```
